mpf/config_spec.yaml

```diff
@@ -331,6 +339,12 @@ fast_coils:
     hold_pwm_patter: single|str|None
 pkone_coils:
     recycle_ms: single|ms|None
+blinkenlight_player:
+    __valid_in__: machine, mode, show
+    __type__: config_player
+    action: single|enum(add,remove,remove_mode,remove_all)|add
+    label: single|str|None
+    color: single|color|ffffff
```

Comment:
This should be changed to:     color: single|str|white
This will allow tokens to be passed in, in addition to color names of color codes.

Reply:
No. This would change to `single|color_or_token|white`

Reply:
Looks like you are right. This does not yet exist. I guess we should create it (as we have it for floats or ints). That will move str parsing out of runtime in most cases.

Reply:
Can't publish changes, but here's what they are.

Update 347 to:     color: single|color_or_token|ffffff.

Then in config_validator:
Add line 81:             "color_or_token": self._validate_type_or_token(self._validate_type_color),
Line 669: Swap above the two lines of comments (#) for the following:
    def _validate_type_color(self, item, validation_failure_info, param=None):
        assert not param
        if item is not None:
            try:
                if isinstance(item, tuple):
                    if len(item) != 3:
                        self.validation_error(item, validation_failure_info, "Color needs three components")
                    return item
            except ValueError:
                self.validation_error(item, validation_failure_info, "Cannot convert value to color.", 11)

Reply:
What's the purpose of `param=None` and `assert not param`?  These aren't in my version of config_validator and I'm not sure the purpose.  I'm not sure what's going on here in general.

Reply:
Jan would have to answer the specifics for you here. But we switched to allowing tokens, and it runs through the token validated to see is it a token on a color, in this case. If it's a token, we know that, so we stop before the validation breaks. If it's not a token, we go through. Without this argument it breaks here as it expects three arguments, but gets 4 due to the token validation.

This is based on the other token_or_xx methods here.

Reply:
Ah ok. I'll take a look when I get home and push the changes later tonight.